dotnet/src/SemanticKernel.Core/Functions/NativeFunction.cs

```diff
@@ -45,6 +46,9 @@ internal sealed class NativeFunction : ISKFunction, IDisposable
     /// <inheritdoc/>
     public string Description { get; }
 
+    /// <inheritdoc/>
+    public ReadOnlyCollection<AIRequestSettings> ModelSettings => Array.Empty<AIRequestSettings>().ToList<AIRequestSettings>().AsReadOnly();
```

Comment:
AFAIK Array.Empty is already a valid ReadOnlyCollection.

```suggestion
    public ReadOnlyCollection<AIRequestSettings> ModelSettings => Array.Empty<AIRequestSettings>();
```

Reply:
Yeah this line is ugly. The suggesting you provided doesn't compile. I'll change to using `IEnumerable` instead.

Reply:
It should compile if you change the property type to IReadOnlyCollection